irods/start_provider.sh

```diff
@@ -23,7 +23,10 @@ python configure_update_collection_mtime_plugin.py
 python configure_unified_storage_tiering_plugin.py
 python configure_users.py
 pkill irodsServer
+pkill irodsReServer
 fi
+
+sleep 10 #give it a bit for the procs to finish
```

Comment:
can we talk about a better solution than just waiting?    can we provide a watcher/checker that loops and sleeps for only 1 sec?

Reply:
yes, how about something like;
```
running=True

while [ $running ]
do
response=$(echo -e "\x00\x00\x00\x33<MsgHeader_PI><type>HEARTBEAT</type></MsgHeader_PI>" | nc localhost 1247)
if [ "${response}" != "HEARTBEAT" ]; then
    running=False
fi
sleep 1
done
```
As per [Monitoring Status of iRODS Servers](https://docs.irods.org/4.2.9/system_overview/tips_and_tricks/)

Reply:
yes, that seems reasonable.

also note that in 4.2.9... the server is now stopped at the end of `setup_irods.py` - so there will need to be an additional `sudo su - irods -c './irodsctl start'` or equivalent to get the system up...  then the HEARTBEAT should be a good way to test that it's listening / alive.

Reply:
I've added a version of this to the MR.

Oddy, when I run it with the `configure_audit_plugin.py` et al tests it seems to hang, but without it runs fine...

Wont the `sudo -iu irods bash -c "cd /usr/sbin; ./irodsServer -u"` on line 40 do the start you mention? It was already there in the 4.2.8 version...